src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,133 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import React, { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [searchParams] = useSearchParams();
+  const Male = 'm';
+  const Female = 'f';
+
+  const preparedPeople = people.map(personValue => ({
+    ...personValue,
+    mother: people.find(human => human.name === personValue.motherName),
+    father: people.find(human => human.name === personValue.fatherName),
+  }));
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(setPeople)
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const handleFilter = () => {
+    let filteredPeople = [...preparedPeople];
```

Comment:
Filter doesn't mutate an array

Reply:
Yeah, but I need `let`, because I rewrite this `filteredPeople ` in `if` for reuse in this function, so I need it✅✅✅